library/core/src/primitive_docs.rs

```diff
@@ -1082,9 +1082,12 @@ impl<T> (T,) {}
 /// bits. Please see [the documentation for [`prim@f32`] or [Wikipedia on
 /// half-precision values][wikipedia] for more information.
 ///
-/// Note that most major platforms will provide `f16` math support by converting to and from
-/// an `f32`, which is usually fairly performant but will not be as fast as using `f32` directly.
-
+/// Note that most common platforms will not support `f16` in hardware without enabling extra target
+/// features, with the notable exception of Apple Silicon (also known as M1/2/etc) processors. For
+/// amd64 this requires the avx512fp16 feature, while RISC-V requires Zhf or Zhfmin, to have hardware support.
```

Comment:
We should probably say x86-64 here? And we may wish to quote the string if we're using the Rust feature name specifically. Also Zhfmin only implements conversions, so,

```suggestion
/// Hardware support on x86-64 requires the avx512fp16 feature, while RISC-V requires Zhf.
```

Alternately we can give a small table?

```suggestion
/// | arch | feature | supported instructions |
/// | --- | --- | --- |
/// | x86_64 | f16c | f16 <-> f32 conversions |
/// | x86_64 | avx512fp16 | f16 arithmetic |
/// | riscv | Zhfmin | f16 <-> float conversions |
/// | riscv | Zhf | f16 arithmetic |
```

Reply:
> We should probably say x86-64 here? And we may wish to quote the string if we're using the Rust feature name specifically. Also Zhfmin only implements conversions, so,
> 
> Alternately we can give a small table?

well heres the issue with quoting off wikipedia lol (Zhfmin), I should probably have read the spec, apologies there

Reply:
yeah, I'm not wild about the table idea, just thought about it.